src/map/lua/lua_baseentity.cpp

```diff
@@ -8625,6 +8625,41 @@ inline int32 CLuaBaseEntity::getAllianceSize(lua_State* L)
     return 1;
 }
 
+/************************************************************************
+*  Function: getAllianceID()
+*  Purpose : Returns the id of the alliance/party or player
+*  Example : local allianceId = player:getAllianceID()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getAllianceID(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+
+    uint8 id = 0;
```

Comment:
someone correct me if I'm wrong, but is 0 a valid alliance ID?

Reply:
I'm not sure, but i can tell you that i have never seen it myself, but it will push nil now to avoid this.